Tasks/DockerV2/task.json

```diff
@@ -13,9 +13,10 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 2,
-        "Minor": 170,
-        "Patch": 2
+        "Minor": 173,
+        "Patch": 0
     },
+    "minimumAgentVersion": "2.172.0",
```

Comment:
This could be a potential breaking changes for people using self hosted agents. Any specific reason for using the latest version?
Is there any new agent specific logic which you've added which this code is dependent on?

Reply:
So before this the container ID was not exposed. That is why this cannot be done with the existing task. There is no way to consistently map between a YAML container alias and it's ID. An updated agent version adds a system to map between those two that this task uses.

Why is this a breaking change? We will wait until that version reaches ring 5 so all customers will have access to that agent.

Reply:
Where are you getting the container id from?

Reply:
See the section of code in dockerlifecycle.ts.
```
    let containerMap = tl.getVariable("agent.containermapping");
```

Reply:
In my understanding it would break the existing pipelines that are configured on older self hosted agents. Is this not correct?

Reply:
>In my understanding it would break the existing pipelines that are configured on older self hosted agents. Is this not correct?

Yup

Instead of restricting the task usage here.. you should probably validate if the variable exists in code and throw an error asking to upgrade the agent.

Reply:
This will not break existing pipelines. The agent automatically updates itself when a task or feature requires a newer version. This is an expected path. We release new YAML features that require updated agents all the time. This PR makes part of this task rely on a new agent feature. If we don’t use the minimum agent field for this, why do we even have it?